ophyd/_pyepics_shim.py

```diff
@@ -134,71 +113,20 @@ def clear_callbacks(self):
 
 
 def release_pvs(*pvs):
-    # Run _release_pvs in the 'monitor' thread, assuring that the CA context is correct
-    def _release_pvs():
-        for pv in pvs:
+    for pv in pvs:
+        pv._reference_count -= 1
+        if pv._reference_count == 0:
             pv.clear_callbacks()
             pv.clear_auto_monitor()
-        event.set()
-
-    event = threading.Event()
-    _dispatcher.get_thread_context('monitor').run(_release_pvs)
-    event.wait()
-
-
-def get_pv(pvname, form='time', connect=False, context=None, timeout=5.0,
-           connection_callback=None, access_callback=None, callback=None,
-           **kwargs):
-    """Get a PV from PV cache or create one if needed.
-
-    Parameters
-    ---------
-    form : str, optional
-        PV form: one of 'native' (default), 'time', 'ctrl'
-    connect : bool, optional
-        whether to wait for connection (default False)
-    context : int, optional
-        PV threading context (defaults to current context)
-    timeout : float, optional
-        connection timeout, in seconds (default 5.0)
-    """
-    if form not in ('native', 'time', 'ctrl'):
-        form = 'native'
-
-    if context is None:
-        context = ca.current_context()
-
-    thispv = None
-
-    # TODO: this needs some work.
-    # thispv = epics.pv._PVcache_.get((pvname, form, context))
-    # if thispv is not None:
-    #     if callback is not None:
-    #         # wrapping is taken care of by `add_callback`
-    #         thispv.add_callback(callback)
-    #     if access_callback is not None:
-    #         access_callback = wrap_callback(_dispatcher, 'metadata',
-    #                                         access_callback)
-    #         thispv.access_callbacks.append(access_callback)
-    #     if connection_callback is not None:
-    #         connection_callback = wrap_callback(_dispatcher, 'metadata',
-    #                                             connection_callback)
-    #         thispv.connection_callbacks.append(connection_callback)
-    #     if thispv.connected:
-    #         if connection_callback:
-    #             thispv.force_connect()
-    #         if access_callback:
-    #             thispv.force_read_access_rights()
-
-    if thispv is None:
-        thispv = PyepicsShimPV(pvname, form=form, callback=callback,
-                               connection_callback=connection_callback,
-                               access_callback=access_callback, **kwargs)
-
-    if connect:
-        if not thispv.wait_for_connection(timeout=timeout):
-            ca.write('cannot connect to %s' % pvname)
-    return thispv
+            if pv.chid is not None:
+                # Clear the channel on the CA-level
+                epics.ca.clear_channel(pv.chid)
+
+            pv.chid = None
+            pv.context = None
+
+            # Ensure we don't get this same PV back again
```

Comment:
Just asking for explanation here.

Is there a bug in `pyepics` where we don't trust reconnect?

Reply:
That is a good question, it does look like pyepics is able to resuscitate a PV that has had it's channel stripped off.  There may have been something funny with the auto-monitor logic?

Seems like good house-keeping to once ophyd is done with a PV to tear it all the way down.....

Reply:
I think going into the cache and wiping itself out will cause issues with anyone who has created `pyepics.PV` separately from the shim. I don't see this as a dealbreaker but think it should at least be mentioned.